libs/@blockprotocol/graph/src/stdlib/subgraph/edge/link-entity.ts

```diff
@@ -205,26 +329,85 @@ export const getRightEntityForLinkEntity = (
  *
  * @param subgraph
  * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
- * @param {Date | string} [timestamp] - An optional `Date` or an ISO-formatted datetime string of the moment to search
- *    for, if not supplied it defaults to the current time
+ * @param {Date | Timestamp} [timestamp] - An optional `Date` or an ISO-formatted datetime string of the moment to
+ *    search for. If the parameter is omitted then results will default to only returning results that are active in
+ *    the latest instant of time in the {@link Subgraph}
  */
 export const getOutgoingLinkAndTargetEntities = <
-  LinkAndRightEntities extends LinkEntityAndRightEntity[] = LinkEntityAndRightEntity[],
+  Temporal extends boolean,
+  LinkAndRightEntities extends LinkEntityAndRightEntity<Temporal>[] = LinkEntityAndRightEntity<Temporal>[],
 >(
-  subgraph: Subgraph,
+  subgraph: Subgraph<Temporal>,
   entityId: EntityId,
-  timestamp?: Date | string,
+  timestamp?: Temporal extends true ? Date | Timestamp : never,
 ): LinkAndRightEntities => {
-  return getOutgoingLinksForEntity(subgraph, entityId, timestamp).map(
-    (linkEntity) => {
-      return {
-        linkEntity,
-        rightEntity: getRightEntityForLinkEntity(
-          subgraph,
+  const searchInterval =
+    timestamp !== undefined
+      ? ({
+          start: {
+            kind: "inclusive",
+            limit: convertTimeToStringWithDefault(timestamp),
+          },
+          end: {
+            kind: "inclusive",
+            limit: convertTimeToStringWithDefault(timestamp),
+          },
```

Comment:
We have this pattern quite often, maybe we should add a function to convert a timestamp into an interval.

Reply:
Sounds sensible, I'll add one

Reply:
Have added one (wasn't actually that many usages :( )

Reply:
([commit](https://github.com/blockprotocol/blockprotocol/pull/928/commits/8c248181676a36894d88ef5ccb23badb1df9bd2e))

Reply:
Thanks, I think it's better, though :)